src/hotspot/cpu/x86/macroAssembler_x86.cpp

```diff
@@ -10421,4 +10421,13 @@ void MacroAssembler::restore_legacy_gprs() {
   movq(rax, Address(rsp, 15 * wordSize));
   addq(rsp, 16 * wordSize);
 }
+
+void MacroAssembler::setcc(Assembler::Condition comparison, Register dst) {
+  if (VM_Version::supports_apx_f()) {
```

Comment:
We could check UseAPX here instead of VM_Version::supports_apx_f().

Reply:
Sounds good, let us keep it this way (VM_Version::supports_*()).

Reply:
Yes, CPU [feature is disabled](https://github.com/openjdk/jdk/commit/a4cf1918c963cbe0b0eee6db580f0769c0cbdbcc#diff-6ed856c57ddbe33e49883adb7c52ec51ed377e5f697dfd6d8bea505a97bfc5a5R1049) if UseAVX is set to false